### PR TITLE
Option to use Data Link in new project

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1161
  - Add context menu option (and associated shortcut) to open a Data Link in a new project

### Important Notes
- The corresponding backend endpoints do not yet exist. The frontend currently passes an additional parameter `dataLinkId: string | null` to the backend's `create_project` endpoint. Note that this is inconsistent with the rest of the backend's terminology which calls Data Links `connector`s, so the parameter name might want to be changed.

### Testing Instructions
- Only the new actions should need to be tested:
  - The "Use In New Project" context menu option on Data Links
  - The associated <kbd>Ctrl</kbd>+<kbd>P</kbd> (<kbd>Cmd</kbd>+<kbd>P</kbd> on macOS) shortcut, when the sole selected asset is a Data Link.
- Minor refactors have been made however, so some other basic testing of the Assets Table is recommended.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectNameColumn.tsx [hunk 1]
```diff
@@ -132,6 +132,7 @@ export default function ProjectNameColumn(props: ProjectNameColumnProps) {
               parentDirectoryId: asset.parentId,
               projectName: asset.title,
               projectTemplateName: event.templateId,
+              dataLinkId: event.dataLinkId,
```
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectNameColumn.tsx]: its `datalinkId`

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectNameColumn.tsx]: should be fixed now

**Comment by PabloBuchu** [comment]
@somebody1234 can we not send `null` values in the body when they are unnecesery when creating a project?

1. When creating from a template:
```
{"parentDirectoryId":"directory-2dMD923ErnhXSBm51ImETf4bcF2","projectName":"Colorado COVID 1","projectTemplateName":"Colorado_COVID","datalinkId":null}
```
`datalinkId` is unnecessary (backend returns 400)
2. From datalink
```
{"parentDirectoryId":"directory-2dMD923ErnhXSBm51ImETf4bcF2","projectName":"asdas 1","projectTemplateName":null,"datalinkId":"connector-2euQX79r7sWOBIq2I5LGMopXEWr"}
```
skip `projectTemplateName`
3. Default
```
{"parentDirectoryId":"directory-2dMD923ErnhXSBm51ImETf4bcF2","projectName":"New Project 1","projectTemplateName":null,"datalinkId":null}
```
remove both `projectTemplateName` and `datalinkId`

**Comment by somebody1234** [comment]
@PabloBuchu should also be done now

**Comment by somebody1234** [comment]
basic testing seems to indicate that the endpoints don't error at the very least

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**